03-Protocol-Overview.md

```diff
@@ -37,18 +37,19 @@ Meanwhile, there are five possible roles (types of software/hardware) for commun
    May optionally provide additional monitoring, receive work from a Job Declarator and use custom work with a pool, or provide other services for a farm.
 
 4. **Job Declarator (optional)**  
-   Receives custom block templates from a Template Provider and declares use of the template with the pool using the Job Declaration Protocol.
-   Further distributes the jobs to Mining Proxy (or Proxies) using the Job Distribution Protocol. This role will often be a built-in part of a Mining Proxy.
+   It is further divided into a Job Declarator Client and a Job Declarator Server.
+   The Job Declarator Client receives custom block templates from a Template Provider and declares use of the template with the Pool or the Job Declarator Server using the Job Declaration Protocol.
+   Further, the Job Declarator Client distributes the jobs to Mining Proxy (or Proxies) using the Job Distribution Protocol. This role will often be a built-in part of a Mining Proxy.
```

Comment:
I would remove this sentence since Job Distribution Protocol has never been specified for real, and we never mentioned it.
```suggestion
```